chrome/content/zotero/xpcom/utilities_internal.js

```diff
@@ -497,7 +497,11 @@ Zotero.Utilities.Internal = {
 				item.linkMode = zoteroItem.attachmentLinkMode;
 				item.mimeType = item.contentType;
 			}
-			
+			// preserve notes
+			if(item.note) { 
+				item.uniqueFields.note = item.note;
+			}
+
```

Comment:
This is for attachment note, right? Then it should be in the `if` statement just above. You can drop this `if` statement. Also, you can drop the "Fix linkMode" comment above too, since it is quite self-explanatory and no longer makes sense.